Lombiq.Tests.UI/Extensions/NavigationUITestContextExtensions.cs

```diff
@@ -204,26 +213,51 @@ public static Task<Uri> GoToSetupPageAndSetupOrchardCoreAsync(this UITestContext
 
     public static async Task<Uri> GoToSetupPageAndSetupOrchardCoreAsync(
         this UITestContext context,
-        OrchardCoreSetupParameters parameters = null)
+        OrchardCoreSetupParameters parameters = null,
+        bool shouldBeSuccess = true)
     {
-        var setupPage = await context.GoToSetupPageAsync(parameters?.RunSetupOnCurrentPage == false);
-        setupPage = await setupPage.SetupOrchardCoreAsync(context, parameters);
+        parameters ??= new(context);
+
+        if (!parameters.RunSetupOnCurrentPage) await context.GoToAbsoluteUrlAsync(context.TestStartUri);
+        await parameters.SetupOrchardCoreAsync(context);
+        context.CheckExistence(OrchardCoreSetupParameters.FinishSetupSelector, !shouldBeSuccess);
 
-        return setupPage.PageUri.Value;
+        return new(context.Driver.Url);
     }
 
+    [Obsolete("Methods using Page<> classes will be removed in the next version.")]
     public static Task<OrchardCoreRegistrationPage> GoToRegistrationPageAsync(this UITestContext context) =>
         context.GoToPageAsync<OrchardCoreRegistrationPage>();
 
+    [Obsolete($"Use {nameof(GoToAdminAsync)} method instead.")]
     public static Task<OrchardCoreDashboardPage> GoToDashboardAsync(this UITestContext context) =>
         context.GoToAdminPageAsync<OrchardCoreDashboardPage>();
 
+    [Obsolete("Methods using Page<> classes will be removed in the next version.")]
     public static Task<OrchardCoreContentItemsPage> GoToContentItemsPageAsync(this UITestContext context) =>
         context.GoToAdminPageAsync<OrchardCoreContentItemsPage>("/Contents/ContentItems");
 
+    [Obsolete("Methods using Page<> classes will be removed in the next version.")]
     public static Task<OrchardCoreFeaturesPage> GoToFeaturesPageAsync(this UITestContext context) =>
         context.GoToAdminPageAsync<OrchardCoreFeaturesPage>("/Features");
 
+    /// <summary>
+    /// Navigate to the Features admin configuration page.
+    /// </summary>
+    public static Task GoToFeaturesAsync(this UITestContext context) =>
+        context.GoToAdminRelativeUrlAsync("/Features");
+
+    /// <summary>
+    /// Navigate to the Features admin configuration page and search for the provided text.
+    /// </summary>
+    /// <returns>A collection of visible checkbox elements.</returns>
+    public static async Task<IEnumerable<IWebElement>> GoToFeaturesAsync(this UITestContext context, string search)
```

Comment:
Use a more specific name, since this is a very specific method (and it's not just going to the Feature screen). But do we really need it here in public? It seems only useful (and is only used) in `TestTurningFeatureOnAndOffAsync`.